packages/superset-ui-legacy-preset-chart-deckgl/src/layers/Geojson/Geojson.jsx

```diff
@@ -136,35 +136,44 @@ const propTypes = {
   setControlValue: PropTypes.func.isRequired,
   viewport: PropTypes.object.isRequired,
   onAddFilter: PropTypes.func,
-  setTooltip: PropTypes.func,
 };
 const defaultProps = {
   onAddFilter() {},
-  setTooltip() {},
 };
 
-function deckGeoJson(props) {
-  const { formData, payload, setControlValue, onAddFilter, setTooltip, viewport } = props;
+class DeckGLGeoJson extends React.Component {
+  containerRef = React.createRef();
 
-  // TODO get this to work
-  // if (formData.autozoom) {
-  //   viewport = common.fitViewport(viewport, geojsonExtent(payload.data.features));
-  // }
-
-  const layer = getLayer(formData, payload, onAddFilter, setTooltip);
+  setTooltip = tooltip => {
+    if (this.containerRef.current) {
+      this.containerRef.current.setTooltip(tooltip);
+    }
+  };
 
-  return (
-    <DeckGLContainer
-      mapboxApiAccessToken={payload.data.mapboxApiKey}
-      viewport={viewport}
-      layers={[layer]}
-      mapStyle={formData.mapbox_style}
-      setControlValue={setControlValue}
-    />
-  );
+  render() {
+    const { formData, payload, setControlValue, onAddFilter, viewport } = this.props;
+
+    // TODO get this to work
```

Comment:
What is autozoom being used for? Has it ever been supported?

Reply:
Seems like a legacy comment from before you touched it,  just asking in case you have context.

Reply:
yes, legacy comment. I also have no idea.